src/main.rs

```diff
@@ -25,21 +27,28 @@ enum Commands {
 enum AppError {
     #[error("cannot connect to database")]
     Connect(#[from] sqlx::Error),
+
+    #[error("cannot start server")]
+    StartServer(#[from] api::ServerError),
 }
 
 async fn daemon(_conn: SqliteConnection) -> Result<(), AppError> {
     info!("starting daemon");
+    api::run_server().await.map_err(AppError::StartServer)?;
+
+    cli_batteries::await_shutdown().await;
+
     Ok(())
 }
 
 async fn app(options: Options) -> Result<(), AppError> {
     // confirm we can use the requested database
-    let conn = SqliteConnection::connect(&options.connection_string).await.map_err(AppError::Connect)?;
+    let conn = SqliteConnection::connect(&options.connection_string)
```

Comment:
We should abstract this away from the DB implementation, you can look at how it is done in signup-sequencer.

Reply:
Yes, absolutely! Database access will be abstracted away so it's not exposing things like whether we're using `sqlite` or `postgres`.

Reply:
Some progress here: https://github.com/worldcoin/tx-sitter/pull/6